### DISCUSSION TITLE
UGS Interface Not Responding After Various Commands

### DISCUSSION THREAD

**Discussion by The-Librarian-369** [winder/Universal-G-Code-Sender#1947] [Help] [answered]
Hello All, I am new to github, UGS, and all things coding/computers. So, if I am not providing essential information, please let me know.

I am running UGS 2.0.11 on a laptop running on Windows 10.
Here is a quick overview of my hardware capabilities, in case it is important.	

Processor	=         11th Gen Intel(R) Core(TM) i3-1115G4 @ 3.00GHz   3.00 GHz
Installed RAM	  =   8.00 GB (7.76 GB usable)
System type	64-bit operating system, x64-based processor

The visualizer works, as well as the editor, command box, and homing action. Overall, using the software is relatively straight-forward with virtually no problems.

However, there is one issue that I have been having trouble resolving. After using the jog controller for a few moments, UGS seems to become largely unresponsive. Almost every button, except for the ones relating to the jog controller, become grey as if they are inactive. My machine moves in the jog state with each button click, but DRO quits responding as well. At this point, my machine will move with jog controls, but nothing else is working. The displayed coordinates (work AND machine) are frozen and the zeroing buttons do not work. None of the usually available actions are responding as well. The homing action, reset zero, return to zero, and soft reset will not work. Pausing/Cancelling File Transfer does nothing. The only thing I am able to do besides jog my machine is disconnect from it. It's almost as if the controller is stuck in jog mode for some reason. 

I think I found a solution, but want to be sure before getting back to my machine. I understand the implemented solution needs to be tested, but I thought maybe someone could at least tell me if this MAY offer resolve.

I followed the file path :
Windows(C:)/user/.../ugs-platform-app-win/etc

From here I opened the file:
ugsplatform.conf

And changed the 64m value to 256m. 

From what I found, posted by BobsCNC, this increases the memory of UGS. I am not sure how, or what anything within that file means, but I'm willing to learn. I am a sponge for knowledge.

Anyhow, would this potentially help me or should I change it back?
Thanks in advance.

**Comment by breiler** [comment, marked as answer]
I think that the setting you have changed will change the amount of initial memory that UGS will use on startup `-Xms=256m`. It should not have any effect on the problem you are describing. There has been some reports of memory problems on Windows but has been caused by large gcode files and the fact that UGS runs in a 32-bit mode which could not utilise all of the system memory.

In this case, another person has reported the same thing and is completely new to me:
https://groups.google.com/g/universal-gcode-sender/c/puiozCgg6bE

If it is a memory problem you could try the latest nightly build for 64-bit windows:
https://github.com/winder/Universal-G-Code-Sender#downloads

**Reply by The-Librarian-369** [reply]
The post in google groups is also me. I have responded there, and will provide further information tonight. Thank you!

I will also attempt using the 64-bit nightly build, and have began learning python to hopefully begin to help in the future! Even though there is a calculator in the setup wizard for UGS, I coded my own calculator for machine calibration just last night. So, win for me! One day, I will be in these message boards helping you help others because I know I greatly appreciate it!!

**Reply by The-Librarian-369** [reply]
So, changing the initial memory used on start-up has not offered resolve, as you had predicted. 

I've taken a few screenshots to hopefully help you visualize the situation better.
I will post them as soon as I figure out why the drag-and-drop feature isn't working for me here.

I will install and use the 64-bit nightly build, and see if that works in the meantime.

**Reply by The-Librarian-369** [reply]
For all of those experiencing an issue similar to the aforementioned, please uninstall UGS and redownload the 64-bit nightly build. This is the fix that was needed for me.
